andela/aragon-cfh
# #141404585 Users should be able to send invites to at most 11 people other than themselves

#### What does this PR do?
Implement restrictions on starting a game (if there aren't enough players) and joining a game (if there are too many players)

#### Description of Task to be completed?
Users should be able to invite people to their game up to a maximum of 11 other people, so no game can exceed 12 people in total and must have at least 3 players.

#### How should this be manually tested?
Run the app and open it in a browser. 
a) On creating a game (or when only one other person has joined a game) try to start the game by clicking the relevant button. A popup informing you that there are too few players should appear.
b) When 11 other people have joined a game, the 12th person to try to join will be redirected to the home page where a popup will explain that the game already has the maximum number of players.
(It can be difficult to mock 12 connections on your own, so you can adjust the playerMaxLimit properties in `config/socket/game.js:30` and `public/js/services/game.js:14` to a more easily testable number.)
c) To test the email invite system, please specify a valid Gmail email and password in your `.env` file (NM_EMAIL and NM_PASSWORD)

#### Any background context you want to provide?
Instead of using a route, I felt using the existing socket framework better fit the design of the app.

#### What are the relevant pivotal tracker stories?
[#141404585](https://www.pivotaltracker.com/n/projects/1986953/stories/141404585)
